taichi/ir/statements.h

```diff
@@ -141,9 +141,16 @@ class BinaryOpStmt : public Stmt {
  public:
   BinaryOpType op_type;
   Stmt *lhs, *rhs;
+  bool is_bit_vectorized;
 
-  BinaryOpStmt(BinaryOpType op_type, Stmt *lhs, Stmt *rhs)
-      : op_type(op_type), lhs(lhs), rhs(rhs) {
+  BinaryOpStmt(BinaryOpType op_type,
+               Stmt *lhs,
+               Stmt *rhs,
+               bool is_bit_vectorized = false)
+      : op_type(op_type),
+        lhs(lhs),
+        rhs(rhs),
+        is_bit_vectorized(is_bit_vectorized) {
```

Comment:
This is a little too much intrusion into the existing system. It seems to me that `is_bit_vectorized` is only used in the `bit_loop_vectorize` pass - maybe you can use an `std::unordered_map<Stmt *, bool>` member variable in `class BitLoopVectorize`, instead of adding a new field in `class BinaryOpStmt`? (Just like `llvm_val` in LLVM codegens.)

Reply:
`is_bit_vectorized` is used only in the `bit_loop_ vectorize` pass when tagged on `BinaryOpStmt` and this part should be replaced with some pass-scope data structure, just as you suggested. But for `GlobalPtrStmt` and `GetChStmt`, they need the tag to pass later passes including `lower_access` and `type_check`

Reply:
Right, that's what I meant: we can use a pass-scope data structure just for `BinaryOpStmt::is_bit_vectorized`. Given we are rushing for the deadline it's fine that we don't do it now.